### PR TITLE
Fix bug in MCI Mapping Importer, and other job/logging updates

### PR DESCRIPTION
**Pull request by gigxz**
Contains https://github.com/greenriver/hmis-warehouse/pull/3358

### CODE REVIEW

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb [hunk 1]
```diff
@@ -83,9 +85,12 @@ def lookup
       parsed = sheet.parse(headers: true)
 
       parsed.each do |row|
+        mci_uniq_id = row['MCI_UNIQ_ID']&.to_i&.to_s
         mci_id = row['MCI_ID']&.to_i&.to_s
+        next unless mci_uniq_id.present? && mci_id.present?
 
-        @lookup[row['MCI_UNIQ_ID']&.to_i&.to_s] = mci_id if mci_id.present?
+        @lookup[mci_uniq_id] = [] unless @lookup.key?(mci_uniq_id)
+        @lookup[mci_uniq_id] << mci_id if mci_id.present?
```
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb]: The bug was that the job only expected one mci id per mci unique id. In fact, the mci unique ids can be mapped to multiple mci ids, and we need to create all of them. I realized this when seeing lots of unrecognized mci ids when importing referrals.

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb [hunk 2]
```diff
@@ -83,9 +85,12 @@ def lookup
       parsed = sheet.parse(headers: true)
 
       parsed.each do |row|
+        mci_uniq_id = row['MCI_UNIQ_ID']&.to_i&.to_s
         mci_id = row['MCI_ID']&.to_i&.to_s
+        next unless mci_uniq_id.present? && mci_id.present?
 
-        @lookup[row['MCI_UNIQ_ID']&.to_i&.to_s] = mci_id if mci_id.present?
+        @lookup[mci_uniq_id] = [] unless @lookup.key?(mci_uniq_id)
```
Comment (ttoomey) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb]: This is totally fine; it's common to use conditional assignment for this kind of thing
`  @lookup[mci_uniq_id] ||= []`

**Review by ttoomey — APPROVED** [review]
Looks great!

**Review by eanders — APPROVED** [review]

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb]: Oh right, thanks :)

**Comment by gigxz** [comment]
merging into https://github.com/greenriver/hmis-warehouse/pull/3358

**Pull request merged**